src/compiler/moduleSpecifiers.ts

```diff
@@ -687,13 +687,17 @@ namespace ts.moduleSpecifiers {
             case Ending.Index:
                 return noExtension;
             case Ending.JsExtension:
-                return noExtension + getJSExtensionForFile(fileName, options);
+                return noExtension + getJSExtensionForFileIfApplicable(fileName, options);
             default:
                 return Debug.assertNever(ending);
         }
     }
 
-    function getJSExtensionForFile(fileName: string, options: CompilerOptions): Extension {
+    function getJSExtensionForFileIfApplicable(fileName: string, options: CompilerOptions): Extension {

```

Comment:
To me this one sounds like the one that never throws. I would suggest naming the undefined-returning one `tryGetJSExtensionForFile` and the throwing one `getJSExtensionForFile`.